apache/trafficcontrol
# Add TO API Routing Blacklist via cdn.conf

## What does this PR (Pull Request) do?
Add a `"routing_blacklist"` field to cdn.conf's "traffic_ops_golang"
section, consisting of two array fields:
- `"perl_routes"`: routes that will be handled by TO-Perl instead of Go
- `"disabled_routes"`: routes that will be disabled by immediately
    returning a 503

These fields are arrays of route IDs (integers). As part of this change,
routes in `traffic_ops_golang` get a unique ID now, as well as a boolean
field indicating whether or not the route can be bypassed to TO-Perl.

Add a `--api-routes` flag to `traffic_ops_golang` that prints out all the
route information (ID, version, path, etc) and exits.


## Which Traffic Control components are affected by this PR?
- Documentation
- Traffic Ops

## What is the best way to verify this PR?
1. Run the `traffic_ops_golang` unit tests
2. Run the TO API tests
3. Configure `cdn.conf` to disable a certain route, make a request to that route, and verify that a 503 response is returned.
4. Configure `cdn.conf` to have Perl handle a certain route, make a request to that route, and verify that the route was handled by TO-Perl and not Go.

## The following criteria are ALL met by this PR
